src/Telegram.Bot/Requests/Stickers/CreateNewStickerSetRequest.cs

```diff
@@ -29,47 +34,82 @@ public abstract class CreateNewStickerSetRequest : FileRequestBase<bool>, IUserT
+    /// </summary>
+    [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+    public StickerType? StickerType { get; set; }
 
     /// <summary>
-    /// An object for position where the mask should be placed on faces
+    /// Pass <see langword="true"/> if stickers in the sticker set must be repainted to the
+    /// color of text when used in messages, the accent color if used as emoji status, white
+    /// on chat photos, or another appropriate color based on context;
+    /// for <see cref="StickerType.CustomEmoji">custom emoji</see> sticker sets only
     /// </summary>
     [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
-    public MaskPosition? MaskPosition { get; set; }
+    public bool? NeedsRepainting { get; set; }
 
     /// <summary>
-    /// Initializes a new request with userId, name and emojis
+    /// Initializes a new request with userId, name, title, stickers and stickerFormat
     /// </summary>
-    /// <param name="userId">User identifier of sticker set owner</param>
+    /// <param name="userId">
+    /// User identifier of sticker set owner
+    /// </param>
     /// <param name="name">
     /// Short name of sticker set, to be used in <c>t.me/addstickers/</c> URLs (e.g., <i>animals</i>).
     /// Can contain only english letters, digits and underscores. Must begin with a letter, can't
     /// contain consecutive underscores and must end in <i>"_by_&lt;bot username&gt;"</i>.
     /// <i>&lt;bot_username&gt;</i> is case insensitive. 1-64 characters
     /// </param>
-    /// <param name="title">Sticker set title, 1-64 characters</param>
-    /// <param name="emojis">One or more emoji corresponding to the sticker</param>
-    protected CreateNewStickerSetRequest(
+    /// <param name="title">
+    /// Sticker set title, 1-64 characters
+    /// </param>
+    /// <param name="stickers">
+    /// A JSON-serialized list of 1-50 initial stickers to be added to the sticker set
+    /// </param>
+    /// <param name="stickerFormat">
+    /// Format of stickers in the set.
+    /// </param>
+    public CreateNewStickerSetRequest(
         long userId,
         string name,
         string title,
-        string emojis)
+        IEnumerable<InputSticker> stickers,
+        StickerFormat stickerFormat)
         : base("createNewStickerSet")
     {
         UserId = userId;
         Name = name;
         Title = title;
-        Emojis = emojis;
+        Stickers = stickers;
+        StickerFormat = stickerFormat;
+    }
+
+    /// <inheritdoc/>
+    public override HttpContent? ToHttpContent()
+    {
+        var multipartContent = GenerateMultipartFormDataContent();
+
+        foreach (var inputSticker in Stickers)
+        {
+            if (inputSticker is InputSticker { Sticker: InputFile file })
+            {
+                multipartContent.AddContentIfInputFile(file, file.FileName!);
+            }
+        }
+
+        return multipartContent;
```

Comment:
shouldn't be there `_ => base.ToHttpContent()` for `InputFileUrl` and `InputFileId` stickers somewhere?

Reply:
I used similar logic as in `SendMediaGroupRequest` class.
I think it is possible if the entire `InputSticker` collection contains everything except `InputFile`.
I don't have a lot of experience with this stuff so not sure.

Reply:
It is possible to do something like this at the beginning of the method:
```C#
if (Stickers.All(inputSticker => inputSticker.Sticker is not InputFile))
    return base.ToHttpContent();
```

Reply:
I think it's the other way around. if you have Any stream/`InputFile` objects in this array, you have to add `MultipartContent` section to request.

Reply:
Like this?
```C#
    public override HttpContent? ToHttpContent()
    {
        if (Stickers.Any(inputSticker => inputSticker.Sticker is InputFile))
        {
            var multipartContent = GenerateMultipartFormDataContent();

            foreach (var inputSticker in Stickers)
            {
                if (inputSticker is InputSticker { Sticker: InputFile file })
                {
                    multipartContent.AddContentIfInputFile(file, file.FileName!);
                }
            }

            return multipartContent;
        }

        return base.ToHttpContent();
    }
```

Reply:
The specification is not clear here. It would be helpful to add test cases for:
- Array of `InputFileId` stickers
- Array of `InputFile` stickers
 - Mixed array with `InputFile` and `InputFileId` stickers.

 to investigate practical possibility to use these combinations.

Reply:
I'd reverse `if` in a fail fast manner:
```suggestion
public override HttpContent? ToHttpContent()
{
    if (!Stickers.Any(inputSticker => inputSticker.Sticker is InputFile))
    {
        return base.ToHttpContent();
    }

    var multipartContent = GenerateMultipartFormDataContent();

    foreach (var inputSticker in Stickers)
    {
        if (inputSticker is InputSticker { Sticker: InputFile file })
        {
            multipartContent.AddContentIfInputFile(file, file.FileName!);
        }
    }

    return multipartContent; 
}
```

Reply:
@karb0f0s Should I apply your recommendations, or like below, will you make changes yourself after the merge?

Reply:
It would be nice to see that this trick actually works with some additional testing. Otherwize there's no need to change anything yet. I'm sure @tuscen will make required changes during PR review/merge.

Reply:
Tests are my weak side so far, this is my first serious experience. There is a creation test for each `StickerFormat`, I'm not sure if this is correct. But as an option, they can be changed according to this recommendation so as not to run into telegram limits.